text/profiles/0239-profiles-data-model.md

```diff
@@ -0,0 +1,1611 @@
+
+### Semantic Conventions
+
+We plan to leverage OTEL Semantic Conventions for various attributes and enums such as profile types or units. Here's a non-exhaustive list of semantic conventions that are used in data model. It is expected to be polished and extended in the future.
+
+#### Attributes
+
+| Attribute  | Type | Description  | Examples  | Requirement Level |
+|---|---|---|---|---|
+| `trace_id` | bytes | Reference to a trace | `0x01020304010203040102030401020304` | Conditionally Required: If `span_id` present. |
+| `span_id` | bytes | Reference to a span | `0x9999999999999999` | Conditionally Required: If `trace_id` present. |
+
+#### Profile Types
+
+Here's a list of possible profile types. It is not exhaustive, and it is expected that more profile types will be added in the future:
+
+* `cpu`
+* `wall`
+* `goroutines`
+* `alloc_objects`
+* `alloc_space`
+* `inuse_objects`
+* `inuse_space`
+* `mutex_contentions`
+* `mutex_delay`
+* `block_contentions`
+* `block_delay`
+
+#### Profile Units
+
+Here's a list of possible profile units. It is not exhaustive, and it is expected that more units will be added in the future. [UCUM](https://ucum.org/) will be used as a reference for some of these units:
+
+* `bytes`
+* `samples`
+* `ns`
+* `ms`
+* `count`
+
+### Decision Log
+
+There were many other alternatives considered during the design process. See [Decision Log](https://github.com/open-telemetry/opentelemetry-proto-profile/blob/54bba7a86d839b9d29488de8e22d8c567d283e7b/opentelemetry/proto/profiles/v1/decision-log.md#L0-L1) for more information about various decisions that were made during the design process.
+
+## Open Questions
+
+Some minor details about the data model are still being discussed and will be flushed out in the future OTEPs. We intend to finalize these details after doing experiments with early versions of working client + collector + backend implementations and getting feedback from the community. The goal of this OTEP is to provide a solid foundation for these experiments and [more](#future-possibilities).
+
+Here's a list of open questions:
+
+### Units in Attributes
+
+Original pprof format allows to specify units for attributes. The current data model supports a similar concept via the use of string table (see `attribute_units` in `Sample` message). It might be a good idea to have the units be specified directly in `KeyValue` message. However, such change would require changes in virtually all signals and it is not clear if it is worth it. We intend to research this question in the future and if it is worth it, we will submit a separate OTEP to make this change.
```

Comment:
I still think unit as metadata makes sense.  If we go that direction, I would want unit as metadata in addition to it being in the metric name.  It isn't a convention in Prometheus to have units as a suffix on attributes, though.  Do you see that as a problem, or are you just making the more general suggestion that units be part of names?

Reply:
I think there are some downsides of having the unit in the metric name as a "/" separated suffix:

1) The unit isn't that useful to see in the UI visualizing a profile since the unit is used to format values and will appear in the value in a scaled form. E.g. seeing "foo/bytes" for values like "1 KiB" would be redundant at best. So tools would have to strip this anyway.
2) It's a fairly implicit convention.